tests/js/spec/components/globalSdkUpdateAlert.test.tsx

```diff
@@ -0,0 +1,278 @@
+import moment from 'moment';
+
+import {mountWithTheme, screen, userEvent} from 'sentry-test/reactTestingLibrary';
+
+import {InnerGlobalSdkUpdateAlert} from 'sentry/components/globalSdkUpdateAlert';
+import {ALL_ACCESS_PROJECTS} from 'sentry/constants/pageFilters';
+import {PageFilters, ProjectSdkUpdates} from 'sentry/types';
+import {DEFAULT_SNOOZE_PROMPT_DAYS} from 'sentry/utils/promptIsDismissed';
+import useApi from 'sentry/utils/useApi';
+import {OrganizationContext} from 'sentry/views/organizationContext';
+
+jest.mock('sentry/utils/useApi');
+
+const makeFilterProps = (filters: Partial<PageFilters>): PageFilters => {
+  return {
+    projects: [1],
+    environments: ['prod'],
+    datetime: {start: new Date(), end: new Date(), period: '14d', utc: true},
+    ...filters,
+  };
+};
+
+const makeSdkUpdateProps = (
+  sdkUpdateProps: Partial<ProjectSdkUpdates>
+): ProjectSdkUpdates[] => {
+  return [
+    {
+      projectId: String(1),
+      sdkName: 'sentry-javascript',
+      sdkVersion: '1.0.0.',
+      suggestions: [
+        {
+          enables: [],
+          newSdkVersion: '1.1.0',
+          sdkName: 'sentry-javascript',
+          type: 'updateSdk',
+        },
+      ],
+      ...sdkUpdateProps,
+    },
+  ];
+};
+
+describe('GlobalSDKUpdateAlert', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('does not shows prompt if projects do not match', async () => {
+    const api = new MockApiClient();
+
+    // We have matching projectId, so updates should be show
+    const filters: PageFilters = makeFilterProps({projects: [1]});
+    const sdkUpdates = makeSdkUpdateProps({projectId: String(1)});
+
+    const promptResponse = {
+      dismissed_ts: undefined,
+      snoozed_ts: undefined,
+    };
+
+    jest.spyOn(api, 'requestPromise').mockResolvedValue({data: promptResponse});
```

Comment:
You can use `MockApiClient.addMockResponse` to set API response values.

Reply:
Pushed a fix, just need to figure out the last spyon case (my comment above)